Lib/test/test_email/test_utils.py

```diff
@@ -48,6 +48,22 @@ def test_parsedate_to_datetime_naive(self):
             utils.parsedate_to_datetime(self.datestring + ' -0000'),
             self.naive_dt)
 
+    def test_parsedate_to_datetime_with_invalid_raises_typeerror(self):
+        with self.assertRaises(TypeError):
```

Comment:
I think this is fine as is, but I think it would be improved if you used a loop and subtests:

```python
invalid_dates = ['', '0', 'A Complete Waste of Time']
for dtstr in invalid_dates:
    with self.subTest(dtstr=dtstr):
        with self.assertRaises(TypeError):
            utils.parsedate_to_datetime(dtstr)
```

Obviously that's pretty nested (though you can probably cut out one layer of nesting by using `self.assertRaises(TypeError, utils.parsedate_to_datetime, dtstr)` rather than the context manager - I still prefer the context manager, myself), so I suppose it's a matter of taste whether DRY and cleaner separation of test cases matters more than keeping code flat.